crates/viewer/src/camera.rs

```diff
@@ -30,16 +47,45 @@ impl Camera {
         self.projection = Projection::Orthographic;
     }
 
+    /// Pan the camera view horizontally and vertically. Look-at target will move along with the
+    /// camera.
+    pub fn pan(&mut self, x: f32, y: f32) {
+        let forward = self.target - self.position;
+        let rightward = self.upward.cross(forward).normalize();
+        let translation = rightward * x + self.upward * y;
+        self.position += translation;
+        self.target += translation;
+    }
+
+    /// Zoom in or out, while looking at the same target.
+    pub fn zoom(&mut self, zoom_delta: f32) {
+        // Change the camera position for perspective projection.
+        let forward = self.target - self.position;
+        self.position += forward * zoom_delta;
+        // Update the zoom factor for orthographic projection.
+        self.zoom_factor = (self.zoom_factor - zoom_delta).max(MIN_ZOOM_FACTOR);
+    }
+
+    /// Orbit around the target while keeping the distance.
+    pub fn rotate(&mut self, yaw: f32, pitch: f32) {
+        let backward = (self.position - self.target).normalize();
+        let leftward = self.upward.cross(backward);
+        let rotation =
+            Quat::from_axis_angle(self.upward, yaw) * Quat::from_axis_angle(leftward, pitch);
+        self.position = (self.target + rotation * backward) * backward.length();
+        self.upward = rotation * self.upward;
+    }
```

Comment:
With the representation suggested above, this could be greatly simplified:
```suggestion
    pub fn rotate(&mut self, rotator: Quat) {
        self.orientation *= rotator;
    }
```